src/bin/commands/install.rs

```diff
@@ -0,0 +1,146 @@
+    /// Do not print
+    #[clap(short, long)]
+    pub quiet: bool,
+    /// Coloring: auto, always, never
+    #[clap(long)]
+    pub color: Option<String>,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+    /// Path to crate
+    #[clap(long)]
+    pub path: Option<String>,
+    /// Path to Cross.toml
+    #[clap(long)]
+    pub config: Option<std::path::PathBuf>,
+    #[clap(name = "crate")]
+    krate: String,
+}
+
+impl Install {
+    pub fn verbose(&self) -> bool {
+        self.verbose
+    }
+
+    pub fn quiet(&self) -> bool {
+        self.quiet
+    }
+
+    pub fn color(&self) -> Option<&str> {
+        self.color.as_deref()
+    }
+
+    pub fn run(self, msg_info: &mut MessageInfo) -> cross::Result<std::process::ExitStatus> {
+        let target_list = rustc::target_list(&mut cross::shell::Verbosity::Quiet.into())?;
+
+        let host_version_meta = rustc::version_meta()?;
+        let mut command = vec!["install".to_owned(), self.krate];
+
+        if let Some(target) = self.target {
+            command.push(format!("--target={target}"));
+        }
+        command.push(format!("--root={}", self.root));
+
+        if let Some(engine) = self.engine {
+            std::env::set_var(docker::CROSS_CONTAINER_ENGINE_VAR, engine);
+        }
+
+        let args = cross::cli::parse(command, &target_list)?;
+        let Some(cross::CrossSetup {
+            config,
+            target,
+            uses_xargo,
+            uses_zig,
+            uses_build_std,
+            zig_version,
+            toolchain,
+            is_remote,
+            engine,
+            image,
+        }) = cross::setup(&host_version_meta, None, &args, target_list, msg_info)? else {
+            eyre::bail!("couldn't setup context for cross (see warning)")
+        };
+
+        let mut is_nightly = toolchain.channel.contains("nightly");
+        let mut rustc_version = None;
+        if let Some((version, channel, _)) = toolchain.rustc_version()? {
+            is_nightly = channel == rustc_version::Channel::Nightly;
+            rustc_version = Some(version);
+        }
+
+        let available_targets = cross::rustup::setup_rustup(&toolchain, msg_info)?;
+
+        cross::rustup::setup_components(
+            &target,
+            uses_xargo,
+            uses_build_std,
+            &toolchain,
+            is_nightly,
+            available_targets,
+            &args,
+            msg_info,
+        )?;
+
+        let filtered_args = cross::get_filtered_args(
+            zig_version,
+            &args,
+            &target,
+            &config,
+            is_nightly,
+            uses_build_std,
+        );
+
+        let cwd = std::env::current_dir()?;
+
+        let paths = docker::DockerPaths::create(
+            &engine,
+            cross::CargoMetadata {
+                workspace_root: cwd.clone(),
+                target_directory: cross::file::absolute_path(self.root)?,
```

Comment:
this is misleading

`cargo install` creates a temporary directory where it builds, we don't use this path as the normal target-dir, rather as the target directory for the `cargo install` output